sensor/kubernetes/enforcer/cronjob/suspend.go

```diff
@@ -0,0 +1,87 @@
+package cronjob
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	kubernetesPkg "github.com/stackrox/rox/pkg/kubernetes"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/sensor/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	batchV1      = "batch/v1"
+	batchV1beta1 = "batch/v1beta1"
+	fieldManager = "StackRox"
+)
+
+// Suspend suspends the cron job
+func Suspend(ctx context.Context, client kubernetes.Interface, deploymentInfo *central.DeploymentEnforcement) (err error) {
+	forcePatch := true
+
+	if ok, apiErr := utils.HasAPI(client, batchV1, kubernetesPkg.CronJob); ok && apiErr == nil {
+		patch := map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name": deploymentInfo.GetDeploymentName(),
+			},
+			"kind":       deploymentInfo.GetDeploymentType(),
+			"apiVersion": batchV1,
+			"spec": map[string]interface{}{
+				"suspend": true,
+			},
+		}
```

Comment:
This now is much closer to `Unstructured`. Not sure what prevented you from just adopting that.

The issue is that **very** similar fragment is repeated below. I don't see a reason for copy-paste if it can be avoided.

E.g. define `makePatch`

```go
func makePatch(deploymentInfo *central.DeploymentEnforcement, apiVersion string) ([]byte, metav1.PatchOptions, error) {
	forcePatch := true

	patch := map[string]interface{}{
		"metadata": map[string]interface{}{
			"name": deploymentInfo.GetDeploymentName(),
		},
		"kind":       deploymentInfo.GetDeploymentType(),
		"apiVersion": apiVersion,
		"spec": map[string]interface{}{
			"suspend": true,
		},
	}
	patchBytes, err := json.Marshal(patch)
	if err != nil {
		return nil, metav1.PatchOptions{}, err
	}

	options := metav1.PatchOptions{
		TypeMeta: metav1.TypeMeta{
			Kind:       deploymentInfo.GetDeploymentType(),
			APIVersion: apiVersion,
		},
		FieldManager: fieldManager,
		Force:        &forcePatch,
	}

	return patchBytes, options, nil
}
```

and use it in `Suspend`:

```go
func Suspend(ctx context.Context, client kubernetes.Interface, deploymentInfo *central.DeploymentEnforcement) (err error) {

	ok, err := utils.HasAPI(client, batchV1, kubernetesPkg.CronJob)
	if err != nil {
		return retry.MakeRetryable(err)
	}

	api := batchV1
	if !ok {
		api = batchV1beta1
	}

	patchBytes, patchOptions, err := makePatch(deploymentInfo, api)
	if err != nil {
		return err
	}

	if ok {
		_, err = client.BatchV1().CronJobs(deploymentInfo.GetNamespace()).Patch(ctx, deploymentInfo.GetDeploymentName(),
			types.ApplyPatchType,
			patchBytes,
			patchOptions)
	} else {
		_, err = client.BatchV1beta1().CronJobs(deploymentInfo.GetNamespace()).Patch(ctx, deploymentInfo.GetDeploymentName(),
			types.ApplyPatchType,
			patchBytes,
			patchOptions)
	}
	if err != nil {
		return retry.MakeRetryable(err)
	}

	return nil
}
```

Reply:
Modified, but not exactly ^